test/Functional/Porter/Net/Http/HttpConnectorTest.php

```diff
@@ -95,10 +103,86 @@ private function stopServer(Process $server)
         $server->stop();
     }
 
+    private function startSsl()
+    {
+        $accept = str_replace($filter = ['[', ']'], null, self::SSL_HOST);
+        $connect = str_replace($filter, null, self::HOST);
+        $certificate = tempnam(sys_get_temp_dir(), 'Porter');
+
+        // Create SSL tunnel process.
+        (new Process(
+            // Generate self-signed SSL certificate in PEM format.
+            "openssl req -new -x509 -nodes -batch -keyout '$certificate' -out '$certificate'
+
+            { stunnel4 -fd 0 || stunnel -fd 0; } <<.
+                # Disable PID to run as non-root user.
+                pid=
+                # Must run as foreground process on Travis, for some reason.
+                foreground=yes
+
+                []
+                cert=$certificate
+                accept=$accept
+                connect=$connect
+."
+        ))->start();
+
+        self::waitForHttpServer(function () {
+            $this->fetchViaSsl(self::createUnverifiedSslConnector());
+        });
+    }
+
+    private static function stopSsl()
+    {
+        `pkill stunnel`;
+    }
+
     private function fetch(Connector $connector = null)
     {
         $connector = $connector ?: $this->connector;
 
         return $connector->fetch('http://' . self::HOST . self::URI);
     }
+
+    private function fetchViaSsl(Connector $connector)
+    {
+        return $connector->fetch('https://' . self::SSL_HOST);
+    }
+
+    /**
+     * Waits for the specified HTTP server invoker to stop throwing HttpConnectionExceptions.
+     *
+     * @param \Closure $serverInvoker HTTP server invoker.
+     */
+    private static function waitForHttpServer(\Closure $serverInvoker)
+    {
+        \ScriptFUSION\Retry\retry(
+            ImportSpecification::DEFAULT_FETCH_ATTEMPTS,
+            $serverInvoker,
+            function (\Exception $exception) {
+                if (!$exception instanceof HttpConnectionException) {
+                    return false;
+                }
+
+                static $handler;
+                $handler = $handler ?: new ExponentialBackoffExceptionHandler;
+
+                return $handler();
+            }
+        );
+    }
+
+    /**
+     * @return HttpConnector
+     */
+    private static function createUnverifiedSslConnector()
+    {
+        $connector = new HttpConnector($options = new HttpOptions);
+        $options->getSslOptions()
+            ->setVerifyPeer(false)
+            ->setVerifyPeerName(false)
```

Comment:
As discussed offline, it might be possible to avoid having to disable SSL verification. This would (probably) involve:

- adding an [OpenSSL root CA generation step](https://datacenteroverlords.com/2012/03/01/creating-your-own-ssl-certificate-authority/),
- signing the user certificate with the root CA key and add the `self::SSL_HOST` ip as a subject and
- passing the root CA certificate filename to `SslOptions::setCertificateAuthorityFilePath`.